src/layers/tile-layer-utils/viewport-utils.test.js

```diff
@@ -0,0 +1,31 @@
+import expect from 'expect';
+import { getTileIndices } from './viewport-utils';
+
+describe('tile-layer-utils/viewport-utils.js', () => {
+  describe('getTileIndices()', () => {
+    it('return a 2 x 2 tile with no projection', () => {
+      const viewport = {
+        zoom: -2,
+        unproject: e => e,
+        height: 8000,
+        width: 8000,
+      };
+      expect(getTileIndices(viewport, 0, -16, 1000)).toEqual([
+        { x: 0, y: 0, z: -2 }, { x: 0, y: 1, z: -2 },
+        { x: 1, y: 0, z: -2 }, { x: 1, y: 1, z: -2 },
+      ]);
+    });
+  });
+
+  describe('getTileIndices()', () => {
```

Comment:
- Just put both `it`s under one `describe`?
- Could you add a couple tests where x/y or height/width are not symmetrical? I think getting them backwards is the kind of error that would be really easy to make, but could be a pain to debug.